sql-db/hibernate-reactive/src/main/java/io/quarkus/ts/reactive/Library.java

```diff
@@ -0,0 +1,113 @@
+package io.quarkus.ts.reactive;
+
+import javax.inject.Inject;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import io.smallrye.mutiny.Multi;
+import io.smallrye.mutiny.Uni;
+
+@Path("/library")
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+public class Library {
+
+    @Inject
+    AuthorRepository authors;
+
+    @GET
+    @Path("books")
+    public Multi<String> all() {
+        return Book.all()
+                .map(Book::getTitle);
+    }
+
+    @GET
+    @Path("books/{id}")
+    public Uni<Response> find(Integer id) {
+        return Book.byId(id)
+                .map(book -> book == null
+                        ? Response.status(Response.Status.NOT_FOUND)
+                        : Response.ok(book.getTitle()))
+                .map(Response.ResponseBuilder::build);
+    }
+
+    @PUT
+    @Path("books/{author}/{name}")
+    public Uni<Response> createBook(Integer author, String name) {
+        Book book = new Book();
+        book.setAuthor(author);
+        book.setTitle(name);
+        return Book.create(author, name)
+                .map(nothing -> Response.status(Response.Status.CREATED))
+                .onFailure().recoverWithItem(error -> Response.status(Response.Status.BAD_REQUEST).entity(error.getMessage()))
+                .map(Response.ResponseBuilder::build);
+    }
+
+    @GET
+    @Path("books/author/{name}")
+    public Multi<String> search(String name) {
+        return authors.byName(name)
+                .flatMap(author -> Multi.createFrom().iterable(author.getBooks()))
+                .map(Book::getTitle);
+    }
+
+    @GET
+    @Path("author")
+    public Multi<String> author() {
+        return authors.streamAll()
+                .map(author -> author.getId() + " " + author.getName());
```

Comment:
I prefer `onItem.tranform` Also, have you think into have a toJson Method in your domain
something like:

```

public String toJsonStringify() {
        return Json.encode(this);
    }
```

And then later you could use it in your endpoints:

```
@GET
    @Path("author")
    public Multi<Response> author() {
        return authors.streamAll()
        .onItem().transform(author -> Response.ok(author.toJsonStringify()).build());
```
something like this...

Reply:
Well, this method is needed only for smoke testing and for debugging purposes, so I see no reason to put much effort into it. Especially, since resteasy is not yet able to produce valid json from Multi anyway[1]

[1] https://github.com/quarkusio/quarkus/issues/18043

Reply:
could you add this issue as a // TODO comment in our method

```
@GET
    @Path("author")
// TODO quarkusio/quarkus#18043
    public Multi<Response> author() {
        return authors.streamAll()
        .onItem().transform(author -> Response.ok(author.toJsonStringify()).build());

```

Reply:
Added the description